Source/Core/InputCommon/ControllerInterface/DInput/XInputFilter.cpp

```diff
@@ -2,129 +2,80 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
-// This function is contained in a separate file because WbemIdl.h pulls in files which break on
-// /Zc:strictStrings, so this compilation unit is compiled without /Zc:strictStrings.
-
-#include <WbemIdl.h>
-#include <Windows.h>
+#include <cwchar>
+#include <unordered_set>
 #include <vector>
 
-#define SAFE_RELEASE(p)                                                                            \
-  {                                                                                                \
-    if (p)                                                                                         \
-    {                                                                                              \
-      (p)->Release();                                                                              \
-      (p) = nullptr;                                                                               \
-    }                                                                                              \
-  }
+// clang-format off
+#include <Windows.h>
+#include <SetupAPI.h>
+// clang-format on
 
 namespace ciface
 {
 namespace DInput
 {
-//-----------------------------------------------------------------------------
-// Modified some MSDN code to get all the XInput device GUID.Data1 values in a vector,
-// faster than checking all the devices for each DirectInput device, like MSDN says to do
-//-----------------------------------------------------------------------------
-void GetXInputGUIDS(std::vector<DWORD>* guids)
+// Code for enumerating hardware devices that use the XINPUT device driver.
+// The MSDN recommended code suffers from massive performance problems on non-English languages,
```

Comment:
I'm not completely clear on the exact nature of the underlying issue.

What I know for sure is that I was easily able to reproduce the problem by installing the Brazilian language pack on my English installation, setting the language to Portuguese in Windows then starting Dolphin. No matter what I did (Use System Language, Use Portuguese (Brazil)) it took much longer to start. Pausing in the debugger revealed it was stuck inside an OLE/COM message loop inside OLE32, sending and receiving data from some external COM server that was taking ages.